packages/ckeditor5-utils/src/observablemixin.ts

```diff
@@ -708,6 +736,35 @@ function attachBindToListeners( observable, toBindings ) {
  * @interface Observable
  * @extends module:utils/emittermixin~Emitter
  */
+export interface Observable extends Emitter {
+	set<K extends keyof this & string>( name: K, value: this[ K ] ): void;
+	set( values: { readonly [ K in keyof this ]?: unknown } ): void;
```

Comment:
I initially tried to `set( values: { readonly [ K in keyof this ]?:` **`this[ K ]`** `} ): void;` It compiled, but in every use site there was an error about incompatible types. I also tried using `Partial<this>` here with the same result. That's why I decided to keep `unknown` here, so the compiler would only check property names, not it's value types. Unfortunately, this style of set method is used a lot in ui-related code.